Library/Homebrew/formula.rb

```diff
@@ -1013,9 +1018,13 @@ def _outdated_versions(options = {})
 
     installed_kegs.each do |keg|
       version = keg.version
+      tab = Tab.for_keg(keg)
       all_versions << version
 
-      return [] if pkg_version <= version && !version.head?
+      latest_version = pkg_version <= version
+      same_version_scheme = version_scheme == tab.version_scheme
+      not_head = !version.head?
+      return [] if latest_version && same_version_scheme && not_head
```

Comment:
And, again, moving `Tab.for_keg(keg)` after `version.head?` so it's not read unnecessariy.